--- a/plugins/wiimote.c	2012-06-13 08:04:20.000000000 -0700
+++ b/plugins/wiimote.c	2012-11-04 01:11:14.795504464 -0700
@@ -56,12 +56,24 @@
  * is pressed.
  */
 
+static uint16_t wii_ids[][2] = {
+       { 0x057e, 0x0306 },
+       { 0x057e, 0x0330 },
+};
+
+static const char *wii_names[] = {
+       "Nintendo RVL-CNT-01",
+       "Nintendo RVL-CNT-01-TR",
+       "Nintendo RVL-WBC-01",
+};
+
 static ssize_t wii_pincb(struct btd_adapter *adapter, struct btd_device *device,
 						char *pinbuf, gboolean *display)
 {
 	uint16_t vendor, product;
 	bdaddr_t sba, dba;
 	char addr[18], name[25];
+	unsigned int i, len;
 
 	adapter_get_address(adapter, &sba);
 	device_get_address(device, &dba, NULL);
@@ -73,14 +85,23 @@
 	device_get_name(device, name, sizeof(name));
 	name[sizeof(name) - 1] = 0;
 
-	if (g_str_equal(name, "Nintendo RVL-CNT-01") ||
-				(vendor == 0x057e && product == 0x0306)) {
-		DBG("Forcing fixed pin on detected wiimote %s", addr);
-		memcpy(pinbuf, &sba, 6);
-		return 6;
-	}
+       len = sizeof(wii_ids) / sizeof(wii_ids[0]);
+       for (i = 0; i < len; ++i) {
+               if (vendor == wii_ids[i][0] && product == wii_ids[i][1])
+                       goto found;
+       }
+
+       len = sizeof(wii_names) / sizeof(wii_names[0]);
+       for (i = 0; i < len; ++i) {
+               if (g_str_equal(name, wii_names[i]))
+                       goto found;
+        }
 
 	return 0;
+found:
+       DBG("Forcing fixed pin on detected wiimote %s", addr);
+       memcpy(pinbuf, adapter_get_address(adapter), 6);
+       return 6;
 }
 
 static int wii_probe(struct btd_adapter *adapter)
